### PR TITLE
Rewrite WPILib Installer Article

### PR DESCRIPTION
**Pull request by Daltz333**
- Uses the new avalonia installer

### CODE REVIEW

File: source/docs/zero-to-robot/step-2/wpilib-setup.rst [hunk 1]
```diff
@@ -1,306 +1,73 @@
...
 
-      Double click on the installer to run it. If you see any Security warnings, click *Run (Windows 7)* or *More Info -> Run Anyway* (Windows 8+).
+.. important:: Due to a signing issue, carefully follow the below steps before continuing on in this document.
 
-      .. image:: images/windows/InstallationType.png
+Please ensure both the installer application and the artifacts ``tar.gz`` is downloaded and placed right next to each other. Additionally, the two files downloaded can **not** be renamed.
 
-      Choose whether to install for *All Users* on the machine or the *Current User*. The *All Users* option requires administrator privileges, but installs in a way that is accessible to all user accounts, the *Current User* install is only accessible from the account it is installed from.
+.. image:: images/wpilib-setup/macos-installer-screenshot.png
+   :alt: Screenshot of the artifact file and installer file next to each other
```
Comment (prateekma) [inline source/docs/zero-to-robot/step-2/wpilib-setup.rst:21]: Delete this, no longer required.

**Review by prateekma — CHANGES_REQUESTED** [review]

File: source/docs/zero-to-robot/step-2/wpilib-setup.rst [hunk 2]
```diff
@@ -1,306 +1,59 @@
...
+   :alt: Installer progress bar
 
-    **What's installed?**
+Congratulations, the WPILib development environment and tooling is now installed on your computer!
 
-      The Offline Installer installs the following components:
+Troubleshooting
+---------------
 
-      -  *Visual Studio Code* - The supported IDE for 2019 and later robot code development. The offline installer sets up a separate copy of VS Code for WPILib development, even if you already have VS Code on your machine. This is done because some of the settings that make the WPILib setup work may break existing workflows if you use VS Code for other projects.
```
Comment (sciencewhiz) [inline source/docs/zero-to-robot/step-2/wpilib-setup.rst:50]: I don't think we should get rid of the description of what's installed

**Review by sciencewhiz — CHANGES_REQUESTED** [review]

File: source/docs/zero-to-robot/step-2/wpilib-setup.rst [hunk 3]
```diff
@@ -1,280 +1,108 @@
...
-      For 2020, the recommended version is 1.41.1 which can be directly downloaded here:  `https://update.code.visualstudio.com/1.41.1/darwin/stable <https://update.code.visualstudio.com/1.41.1/darwin/stable>`__
+You can download the latest release of the installer from `GitHub <https://github.com/wpilibsuite/allwpilib/releases/latest/>`__. Ensure that you download the correct binary for your OS and architecture.
 
-      The downloaded file will be "VSCode-darwin-stable.zip" (1)
+Extracting the Installer
+------------------------
 
-      Once downloaded, double-click on the zip file to expand it and copy the
-      new file: "Visual Studio Code" to the Applications folder (2).
+When you download the WPILib installer, it is distributed as a disk image file ``.iso`` for Windows and Linux. It is distributed as a DMG for MacOS.
```
Comment (sciencewhiz) [inline source/docs/zero-to-robot/step-2/wpilib-setup.rst:14]: it's a tar.gz for linux

File: source/docs/zero-to-robot/step-2/wpilib-setup.rst [hunk 4]
```diff
@@ -297,10 +125,14 @@ WPILib is designed to install to different folders for different years, so that
...
   .. tab:: macOS
 
-     1. Delete Visual Studio Code from the Applications folder. Alternately, the extensions can be `uninstalled <https://code.visualstudio.com/docs/editor/extension-gallery#_manage-extensions>`__
-     2. Delete the appropriate wpilib folder (2019: ``~/frc2019``, 2020 and later: ``~/wpilib/YYYY`` where ``YYYY`` is the year to uninstall)
+     1. Run ``rm -rf ~/wpilib/2021`` from a terminal.
 
   .. tab:: Linux
 
      1. Delete the appropriate wpilib folder (2019: ``~/frc2019``, 2020 and later: ``~/wpilib/YYYY`` where ``YYYY`` is the year to uninstall). eg ``rm -rf ~/frc2019``
      2. Uninstall Visual Studio Code ``sudo dpkg -r code``. Alternately, the extensions can be `uninstalled <https://code.visualstudio.com/docs/editor/extension-gallery#_manage-extensions>`__
```
Comment (sciencewhiz) [inline source/docs/zero-to-robot/step-2/wpilib-setup.rst:133]: I lied, uninstall instructions for linux aren't the same. You can delete step 2.

File: source/docs/zero-to-robot/step-2/wpilib-setup.rst [hunk 5]
```diff
@@ -297,10 +125,14 @@ WPILib is designed to install to different folders for different years, so that
 
   .. tab:: macOS
 
-     1. Delete Visual Studio Code from the Applications folder. Alternately, the extensions can be `uninstalled <https://code.visualstudio.com/docs/editor/extension-gallery#_manage-extensions>`__
-     2. Delete the appropriate wpilib folder (2019: ``~/frc2019``, 2020 and later: ``~/wpilib/YYYY`` where ``YYYY`` is the year to uninstall)
+     1. Run ``rm -rf ~/wpilib/2021`` from a terminal.
```
Comment (sciencewhiz) [inline source/docs/zero-to-robot/step-2/wpilib-setup.rst:128]: I think the previous step 2 is more user friendly then opening a terminal

File: source/docs/zero-to-robot/step-2/wpilib-setup.rst [hunk 6]
```diff
@@ -297,10 +125,14 @@ WPILib is designed to install to different folders for different years, so that
...
 
   .. tab:: Linux
 
      1. Delete the appropriate wpilib folder (2019: ``~/frc2019``, 2020 and later: ``~/wpilib/YYYY`` where ``YYYY`` is the year to uninstall). eg ``rm -rf ~/frc2019``
```
Comment (sciencewhiz) [inline source/docs/zero-to-robot/step-2/wpilib-setup.rst:132]:
```suggestion
     1. Delete the appropriate wpilib folder (2019: ``~/frc2019``, 2020 and later: ``~/wpilib/YYYY`` where ``YYYY`` is the year to uninstall). eg ``rm -rf ~/wpilib/2021``
```

**Review by sciencewhiz — DISMISSED** [review]

File: source/docs/software/vscode-overview/3rd-party-libraries.rst [hunk 7]
```diff
@@ -89,10 +89,3 @@ Removing a Library Dependency
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
 To remove a library dependency from a project, select **Manage Current Libraries** from the **Manage Vendor Libraries** menu, check the box for any libraries to uninstall and click OK. These libraries will be removed as dependencies from the project.
-
```
Comment (sciencewhiz) [inline source/docs/software/vscode-overview/3rd-party-libraries.rst:92]: Is removing this intentional?

File: source/docs/zero-to-robot/step-2/wpilib-setup.rst [hunk 8]
```diff
@@ -1,280 +1,108 @@
 WPILib Installation Guide
-=============================
+=========================
 
-.. note:: This setup is for C++/Java teams. LabVIEW teams should proceed to :doc:`Installing LabVIEW for FRC </docs/zero-to-robot/step-2/labview-setup>`.
+This guide is intended for Java and C++ teams. LabVIEW teams can skip to :doc:`labview-setup`. Additionally, the below tutorial shows Windows 10, but the steps are identical for all OSes. Notes differentating OSes will be shown.
```
Comment (AustinShalit) [inline source/docs/zero-to-robot/step-2/wpilib-setup.rst:4]: I think the last 2 sentences are awkward.

I would not use an initialism for `OS`.

File: source/docs/zero-to-robot/step-2/wpilib-setup.rst [hunk 9]
```diff
@@ -1,280 +1,108 @@
...
-=============================
+=========================
 
-.. note:: This setup is for C++/Java teams. LabVIEW teams should proceed to :doc:`Installing LabVIEW for FRC </docs/zero-to-robot/step-2/labview-setup>`.
+This guide is intended for Java and C++ teams. LabVIEW teams can skip to :doc:`labview-setup`. Additionally, the below tutorial shows Windows 10, but the steps are identical for all operating systems. Notes differentating operating systems will be shown.
 
-.. tabs::
-
-  .. tab:: Windows
-    .. note:: Windows 7: You **must** install the NI FRC Game Tools or .NET Version 4.62 (or later) before proceeding with the install of Visual Studio Code for FRC. The NI FRC Game Tools installer will automatically install the proper version of .NET. The standalone .NET installer is `here <https://support.microsoft.com/en-us/help/3151800/the-net-framework-4-6-2-offline-installer-for-windows>`__
```
Comment (sciencewhiz) [inline source/docs/zero-to-robot/step-2/wpilib-setup.rst:9]: I'm guessing this is still needed, unless Avalonia needs an even later version of .NET

Comment (Daltz333) [reply source/docs/zero-to-robot/step-2/wpilib-setup.rst:9]: Resolved in 9140e1c

**Comment by sciencewhiz** [comment]
@Kevin-OConnor do you still have a Windows 7 computer to test with?

**Comment by sciencewhiz** [comment]
I installed on a Windows 7 SP1 VM that was last updated in 2015. I only had to install the listed update.

**Review by sciencewhiz — APPROVED** [review]

**Pull request merged**